test/fit_tests/common/fit_common.py

```diff
@@ -0,0 +1,631 @@
+    global API_PROTOCOL
+    global API_PORT
+
+    if API_PROTOCOL == "None":
+        if API_PORT == "None":
+            API_PORT = str(GLOBAL_CONFIG['ports']['http'])
+        if restful("http://" + ARGS_LIST['ora'] + ":" + str(API_PORT) + "/", rest_timeout=5)['status'] == 0:
+            API_PROTOCOL = 'https'
+            API_PORT = str(GLOBAL_CONFIG['ports']['https'])
+        else:
+            API_PROTOCOL = 'http'
+            API_PORT = str(GLOBAL_CONFIG['ports']['http'])
+
+    # Retrieve authentication token for the session
+    if AUTH_TOKEN == "None":
+        get_auth_token()
+
+    return restful(API_PROTOCOL + "://" + ARGS_LIST['ora'] + ":" + str(API_PORT) + url_cmd,
+                       rest_action=action, rest_payload=payload, rest_timeout=timeout, rest_headers=headers)
+
+def restful(url_command, rest_action='get', rest_payload=[], rest_timeout=None, sslverify=False, rest_headers={}):
+    '''
+    This routine executes a rest API call to the host.
+
+    :param url_command: the full URL for the command
+    :param rest_action: what the restful do (get/post/put/delete)
+    :param rest_payload: payload for rest request
+    :param rest_headers: headers (JSON dict)
+    :param rest_timeout: timeout for rest request
+    :param sslverify: ssl Verify (True/False)
+    :return:    {'json':result_data.json(), 'text':result_data.text,
+                'status':result_data.status_code,
+                'headers':result_data.headers,
+                'timeout':False}
+    '''
+
+    result_data = None
+
+    # print URL and action
+    if VERBOSITY >= 4:
+        print "restful: Action = ", rest_action, ", URL = ", url_command
+
+    # prepare payload for XML output
+    payload_print = []
+    try:
+        json.dumps(rest_payload)
+    except:
+        payload_print = []
+    else:
+        payload_print = json.dumps(rest_payload, sort_keys=True, indent=4,)
+        if len(payload_print) > 4096:
+            payload_print = payload_print[0:4096] + '\n...truncated...\n'
+        if VERBOSITY >= 9 and rest_payload != []:
+            print "restful: Payload =\n", payload_print
+
+    rest_headers.update({"Content-Type": "application/json"})
+    if VERBOSITY >= 5:
+         print "restful: Request Headers =", rest_headers, "\n"
+
+    # If AUTH_TOKEN is set, add to header
+    if AUTH_TOKEN != "None" and AUTH_TOKEN != "Unavailable" and "authorization" not in rest_headers:
+        rest_headers.update({"authorization": "JWT " + AUTH_TOKEN, "X-Auth-Token": REDFISH_TOKEN})
+    # Perform rest request
+    try:
+        if rest_action == "get":
+            result_data = requests.get(url_command,
+                                       timeout=rest_timeout,
+                                       verify=sslverify,
+                                       headers=rest_headers)
+        if rest_action == "delete":
+            result_data = requests.delete(url_command,
```

Comment:
For some API such as rack node relation ship management, it requires send json payload in http delete method. Can you add  data=json.dumps(rest_payload), in the request.delete call?